enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,167 @@
+
+## Motivation
+
+### Goals
+As an administrator, I would like to create or reuse my own networks, subnets and routing schemes that I can deploy OpenShift to.
+
+### Non-Goals
+There are no changes in expectations regarding publicly addressable parts of the cluster.
+
+## Proposal
+- Installer allow users to provide a list of subnets that should be used for the cluster. 
+- In order to support provider networks in IPI, the subnets passed to the installer must meet these requirements:
+  - have the capacity and ability for the installer to provision ports on the nodes subnet
+  - have dhcp enabled
+  - the installer must be able to send and recieve https traffic to nodes connected to ports on the nodes subnet
+- In order to support provider networks in UPI, additional documentation will be provided to support this use case
+- For those who do not want the installer to provision their networking resources, we will provide additional UPI documentation.
+- The infrastructure resources owned by the cluster continue to be clearly identifiable and distinct from other resources.
+- Destroying a cluster must make sure that no resources are deleted that didn't belong to the cluster. Leaking resources is preferred over any possibility of deleting non-cluster resources.
+
+
+## User Stories
+1. As an administrator of a private openstack cloud, I use provider networks for my openstack cluster, and would like to be able to install OpenShift in this environment.
+2. As an OpenStack tenant, I want to be able to use pre-configured network resources so that I dont have to create new resources to install an OpenShift cluster.
+
+## Implementation Details Notes/Constraints
+
+### Installer Resources
+If a user wants to use a pre-existing subnet, they can provide the installer with the ID of that subnet. The installer needs to have the authorization and quota to provision the necessary ports and resources on that subnet.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    nodesSubnet: abcd-4321
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+Users with provider networks often do not use floating IPs, and so the first changge we will make to address this is to make the parameter `platform.openstack.lbFloatingIP` optional. When unset, the installer will not attempt to provision and attach a floating ip to the api port. To allow the customer to set up their own custom external access infrastructure, we will have to let them set the `apiVIP` and `ingressVIP` port IPs ahead of the install, so that they can be added to their routing/loadbalancing scheme before the installer is run. 
```

Comment:
s/changge/change/

Small correction: the installer doesn't provision a floating IP for the the API (it must exist before running the installer), but only attaches it to API VIP port.

It makes sense to allow setting of apiVIP and ingressVIP so that the installer stops using the hardcoded `.5` and `.7` addresses. We'll also want to drop the DNS VIP similarly to BM.